File: src/main/java/kitchenpos/menus/tobe/domain/DisplayedName.java
package kitchenpos.menus.tobe.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import jakarta.persistence.Embedded;


@Embeddable
public class DisplayedName {
    @Column(name = "displayed", nullable = false)
    private boolean displayed;

    @Embedded
    private CleanName cleanName;

    protected DisplayedName() {}

    public DisplayedName(boolean displayed, String name) {
        this(displayed, new CleanName(name));
    }

    public DisplayedName(boolean displayed, String name, MenuNameValidationService menuNameValidationService) {
        this(displayed, new CleanName(name, menuNameValidationService));
    }

    public DisplayedName(boolean displayed, CleanName cleanName) {
        this.displayed = displayed;
        this.cleanName = cleanName;
    }

    public void display() {
        displayed = true;
    }

    public void hide() {
        displayed = false;
    }

    public boolean isDisplayed() {
        return displayed;
    }
}

File: src/main/java/kitchenpos/menus/tobe/domain/DisplayedName.kt
package kitchenpos.menus.tobe.domain

class DisplayedName(val value: String, blackWords: BlackWords) {
    init {
        require(value.isNotEmpty()) { "메뉴 그룹 이름은 비어있을 수 없습니다." }
        require(!blackWords.contains(value)) { "금지어가 포함되어 있습니다." }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is DisplayedName) return false

        if (value != other.value) return false

        return true
    }

    override fun hashCode(): Int {
        return value.hashCode()
    }
}
